src/devices/cpu/axc51/axc51.cpp

```diff
@@ -98,166 +119,155 @@ device_memory_interface::space_config_vector axc51base_cpu_device::memory_space_
 	};
 }
 
-
-/***************************************************************************
-    MACROS
-***************************************************************************/
-
-/* Read Opcode/Opcode Arguments from Program Code */
-#define ROP(pc)         m_program.read_byte(pc)
-#define ROP_ARG(pc)     m_program.read_byte(pc)
-
-/* Read a byte from External Code Memory (Usually Program Rom(s) Space) */
-#define CODEMEM_R(a)    (uint8_t)m_program.read_byte(a)
-
-/* Read/Write a byte from/to External Data Memory (Usually RAM or other I/O) */
-#define DATAMEM_R(a)    (uint8_t)m_io.read_byte(a)
-#define DATAMEM_W(a,v)  m_io.write_byte(a, v)
-
-/* Read/Write a byte from/to the Internal RAM */
-
-#define IRAM_R(a)       iram_read(a)
-#define IRAM_W(a, d)    iram_write(a, d)
-
 /* Read/Write a byte from/to the Internal RAM indirectly */
 /* (called from indirect addressing)                     */
-uint8_t axc51base_cpu_device::iram_iread(offs_t a) { return m_data.read_byte(a); }
-void axc51base_cpu_device::iram_iwrite(offs_t a, uint8_t d) { m_data.write_byte(a, d); }
-
-#define IRAM_IR(a)      iram_iread(a)
-#define IRAM_IW(a, d)   iram_iwrite(a, d)
-
-/* Form an Address to Read/Write to External RAM indirectly */
-/* (called from indirect addressing)                        */
-#define ERAM_ADDR(a,m)  external_ram_iaddr(a,m)
-
-/* Read/Write a bit from Bit Addressable Memory */
-#define BIT_R(a)        bit_address_r(a)
-#define BIT_W(a,v)      bit_address_w(a, v)
-
+/* these go through DBASE register on axc51 (at least stack accesses) */
+uint8_t axc51base_cpu_device::iram_indirect_read(offs_t a) { return m_data.read_byte((m_sfr_regs[AXC51_DBASE - 0x80] * 4) + a); }
+void axc51base_cpu_device::iram_indirect_write(offs_t a, uint8_t d) { m_data.write_byte((m_sfr_regs[AXC51_DBASE - 0x80] * 4) + a, d); }
 
 /***************************************************************************
     SHORTCUTS
 ***************************************************************************/
 
-#define PPC     m_ppc
-#define PC      m_pc
-
 /* SFR Registers - These are accessed directly for speed on read */
 /* Read accessors                                                */
 
-#define SFR_A(a)        m_sfr_ram[(a)]
+#define SFR_A(a)        m_sfr_regs[(a-0x80)]
```

Comment:
That's scary, could you at least mask it like array[(a) & 0x7f]?
From what I saw the array size is 0x80.

Reply:
They're only used with the defined constants I believe, I'm more inclined
just to chop 0x80 off those and add a note that they're based at 0x80,
rather than them being the absolute address.

Reply:
ok reworked this anyway, and nuked the macro